pgcontents/tests/test_synchronization.py

```diff
@@ -177,3 +184,99 @@ def check_reencryption(old, new):
         # crypto manager.
         unencrypt_all_users(engine, crypto2_factory, logger)
         check_reencryption(manager2, no_crypto_manager)
+
+
+class TestGenerateNotebooks(TestCase):
+
+    def setUp(self):
+        remigrate_test_schema()
+
+    def tearDown(self):
+        clear_test_db()
+
+    def test_generate_notebooks(self):
+        """
+        Creates files and checkpoints for two users and checks their
+        accessibility through `generate_files` and `generate_checkpoints`.
+        """
+        db_url = TEST_DB_URL
+        engine = create_engine(db_url)
+        user_ids = ['test_generate_notebooks0', 'test_generate_notebooks1']
+        encryption_pw = u'foobar'
+
+        crypto_factory = single_password_crypto_factory(encryption_pw)
+        managers = {}
+        for user_id in user_ids:
+            managers[user_id] = PostgresContentsManager(
+                user_id=user_id,
+                db_url=db_url,
+                crypto=crypto_factory(user_id),
+                create_user_on_startup=True,
+            )
+        paths = {}
+
+        def create_nbs(user_id):
+            """
+            Create test notebooks and create a checkpoint for each.
+            """
+            paths[user_id] = set(populate(managers[user_id]))
+            for path in paths[user_id]:
+                managers[user_id].create_checkpoint(path)
+
+        # Create test files/remote checkpoints for both users, their creation
+        # times separated by a recorded datetime.
+        create_nbs(user_ids[0])
+        split_dt = datetime.now()
```

Comment:
I think there's a possibility for a narrow race condition here.  We're comparing the result of `datetime.now()` to the the database's internal clock, but there's no guarantee that those are exactly synchronized.  Safer here would probably be to use the last modified date on the notebooks written for the first user.